Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp

```diff
@@ -793,6 +813,33 @@ namespace AssetProcessor
         QueueIdleCheck();
     }
 
+    bool AssetProcessorManager::IsLfsPointerFile(const AZStd::string& filePath)
+    {
+        if (!m_lfsPointerFileValidator)
+        {
+            AZStd::vector<AZStd::string> scanDirectories;
+            if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
+            {
+                scanDirectories.emplace_back(AZ::Utils::GetEnginePath(settingsRegistry).c_str());
+                scanDirectories.emplace_back(AZ::Utils::GetProjectPath(settingsRegistry).c_str());
+
+                auto RetrieveActiveGemRootDirectories = [&scanDirectories](AZStd::string_view, AZStd::string_view gemPath)
+                {
+                    scanDirectories.emplace_back(gemPath.data());
+                };
+                AZ::SettingsRegistryMergeUtils::VisitActiveGems(*settingsRegistry, RetrieveActiveGemRootDirectories);
```

Comment:
This looks real re-usable.
Maybe this should be a helper function... FindScanFolders() or something?

Reply:
Sure. I can make a helper function for it.